src/ansys/dpf/composites/_composite_model_impl.py

```diff
@@ -0,0 +1,613 @@
+            )
+
+            minmax_el_op.inputs.mesh(self.get_mesh())
+            minmax_el_op.inputs.material_support(
+                self.material_operators.material_support_provider.outputs
+            )
+
+            if has_layup_provider and write_data_for_full_element_scope:
+                add_default_data_op = dpf.Operator("composite::add_default_data")
+                add_default_data_op.inputs.requested_element_scoping(chunking_generator.outputs)
+                add_default_data_op.inputs.time_id(
+                    evaluate_failure_criterion_per_scope_op.outputs.time_id
+                )
+
+                add_default_data_op.inputs.mesh(self.get_mesh())
+
+                add_default_data_op.inputs.fields_container(minmax_el_op.outputs.field_min)
+                add_default_data_op.run()
+
+                add_default_data_op.inputs.fields_container(minmax_el_op.outputs.field_max)
+                add_default_data_op.run()
+
+            # It is important to evaluate the field here, otherwise the merge operator detects
+            # the workflow as changed if upstream operator inputs change
+            min_container = minmax_el_op.outputs.field_min()
+            max_container = minmax_el_op.outputs.field_max()
+
+            converter_op = dpf.Operator("composite::failure_measure_converter")
+            converter_op.inputs.fields_container(min_container)
+            converter_op.inputs.measure_type(measure.value)
+            converter_op.run()
+
+            converter_op.inputs.fields_container(max_container)
+            converter_op.run()
+
+            min_merger.connect(merge_index, min_container)
+            max_merger.connect(merge_index, max_container)
+            merge_index = merge_index + 1
+
+        if merge_index == 0:
+            raise RuntimeError(
+                "No output is generated! Please check the scope (element and ply ids)."
+            )
+
+        if measure == FailureMeasureEnum.INVERSE_RESERVE_FACTOR:
+            return max_merger.outputs.merged_fields_container()
+        else:
+            return min_merger.outputs.merged_fields_container()
+
+    @_deprecated_composite_definition_label
+    def get_sampling_point(
+        self,
+        combined_criterion: CombinedFailureCriterion,
+        element_id: int,
+        time: Optional[float] = None,
+        composite_definition_label: Optional[str] = None,
+    ) -> SamplingPointNew:
+        """Get a sampling point for an element ID and failure criteria.
+
+        Parameters
+        ----------
+        combined_criterion:
+            Combined failure criterion to evaluate.
+        element_id:
+            Element ID or label of the sampling point.
+        time:
+            Time or frequency at which to evaluate the sampling point. If ``None``,
+            the last time or frequency in the result file is used.
+        composite_definition_label:
```

Comment:
```suggestion
        composite_definition_label: str, optional
```

Reply:
The type will be automatically added from the python type hint